tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -247,7 +246,7 @@ peekFieldName = do
   if fieldName == nullPtr then
     pure Nothing
   else
-    Just . FieldName . camelCase <$> liftIO (peekCString fieldName)
+    Just . FieldName . toHaskellCamelCaseIdentifier <$> liftIO (peekCString fieldName)
```

Comment:
This was an important fix for reserved names in haskell like `type` which get a prime appended to them.